zerver/lib/upload.py

```diff
@@ -459,6 +493,25 @@ def ensure_medium_avatar_image(self, email):
         resized_medium = resize_avatar(image_data, MEDIUM_AVATAR_SIZE)
         write_local_file('avatars', file_path + '-medium.png', resized_medium)
 
+    def upload_emoji_image(self, emoji_file, emoji_file_name, user_profile):
+        # type: (File, Text, UserProfile) -> None
+        emoji_path = RealmEmoji.PATH_ID_TEMPLATE.format(
+            realm_id= user_profile.realm_id,
+            emoji_file_name=emoji_file_name
+        )
+
+        image_data = emoji_file.read()
+        write_local_file(
+            'avatars',
+            emoji_path,
+            image_data)
+
+    def get_emoji_url(self, emoji_file_name, realm_id):
+        # type: (Text, int) -> Text
+        return os.path.join(
+            u"/user_avatars",
+            RealmEmoji.PATH_ID_TEMPLATE.format(realm_id=realm_id, emoji_file_name=emoji_file_name))
```

Comment:
@timabbott, @rishig I can offer to add Wand library, which is binding to ImageMagic, but it requires to install imageMagic and Wand dev library to the system. This will simplify processing animated images. Pillow and Pil can't resize gif files, It just can collect files to one on re-save gif files.